server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -1023,6 +1023,16 @@ public boolean isSegRepEnabled() {
         return ReplicationType.SEGMENT.equals(replicationType);
     }
 
+    public boolean isSegRepLocalEnabled() {
+        return isSegRepEnabled() && !isSegRepWithRemoteEnabled();
+    }
+
+    public boolean isSegRepWithRemoteEnabled() {
+        return isSegRepEnabled()
+            && isRemoteStoreEnabled()
+            && FeatureFlags.isEnabled(FeatureFlags.SEGMENT_REPLICATION_WITH_REMOTE_STORE_EXPERIMENTAL);
```

Comment:
As noted above, I think we need a new setting here to differentiate using segrep side by side with remote store and using remote store as the src of replication.  The new setting can be gated with `SEGMENT_REPLICATION_EXPERIMENTAL`